akka-http-tests/src/test/scala/akka/http/scaladsl/marshalling/ContentNegotiationSpec.scala

```diff
@@ -13,83 +16,88 @@ import MediaTypes._
 
-    "(without headers)" in test { accept ⇒
-      accept(`text/plain`) should select(`text/plain`, `UTF-8`)
+    "(without headers)" test { accept ⇒
+      //accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/plain` withCharset `UTF-16`) should select(`text/plain`, `UTF-16`)
     }
 
-    "Accept: */*" in test { accept ⇒
+    "Accept: */*" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/plain` withCharset `UTF-16`) should select(`text/plain`, `UTF-16`)
     }
 
-    "Accept: */*;q=.8" in test { accept ⇒
+    "Accept: */*;q=.8" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/plain` withCharset `UTF-16`) should select(`text/plain`, `UTF-16`)
     }
 
-    "Accept: text/*" in test { accept ⇒
+    "Accept: text/*" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/xml` withCharset `UTF-16`) should select(`text/xml`, `UTF-16`)
       accept(`audio/ogg`) should reject
     }
 
-    "Accept: text/*;q=.8" in test { accept ⇒
+    "Accept: text/*;q=.8" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/xml` withCharset `UTF-16`) should select(`text/xml`, `UTF-16`)
       accept(`audio/ogg`) should reject
     }
 
-    "Accept: text/*;q=0" in test { accept ⇒
+    "Accept: text/*;q=0" test { accept ⇒
       accept(`text/plain`) should reject
       accept(`text/xml` withCharset `UTF-16`) should reject
       accept(`audio/ogg`) should reject
     }
 
-    "Accept-Charset: UTF-16" in test { accept ⇒
+    "Accept-Charset: UTF-16" test { accept ⇒
+      accept(`text/plain`) should select(`text/plain`, `UTF-16`)
+      accept(`text/plain` withCharset `UTF-8`) should reject
+    }
+
+    "manually created Accept-Charset: UTF-16" in testHeaders(headers.`Accept-Charset`(Vector(HttpCharsets.`UTF-16`.toRange))) { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-16`)
       accept(`text/plain` withCharset `UTF-8`) should reject
     }
 
-    "Accept-Charset: UTF-16, UTF-8" in test { accept ⇒
+    "Accept-Charset: UTF-16, UTF-8" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-8`)
       accept(`text/plain` withCharset `UTF-16`) should select(`text/plain`, `UTF-16`)
     }
 
-    "Accept-Charset: UTF-8;q=.2, UTF-16" in test { accept ⇒
+    "Accept-Charset: UTF-8;q=.2, UTF-16" test { accept ⇒
       accept(`text/plain`) should select(`text/plain`, `UTF-16`)
       accept(`text/plain` withCharset `UTF-8`) should select(`text/plain`, `UTF-8`)
     }
 
-    "Accept-Charset: UTF-8;q=.2" in test { accept ⇒
-      accept(`text/plain`) should select(`text/plain`, `ISO-8859-1`)
+    "Accept-Charset: UTF-8;q=.2" test { accept ⇒
+      // FIXME: which rule would enforce that behavior?
+      // accept(`text/plain`) should select(`text/plain`, `ISO-8859-1`)
```

Comment:
Sorry for being way late to the party here, but I'd like to shine some light on this test and hopefully clear up the confusion.
This test was originally written for [section 14.2 of RFC 2616](http://tools.ietf.org/html/rfc2616#section-14.2), which says:

> If no "*" is present in an Accept-Charset field, then all character sets not explicitly mentioned get a quality value of 0, except for ISO-8859-1, which gets a quality value of 1 if not explicitly mentioned.

However, in the latest specs `ISO-8859-1` doesn't have this special status anymore and the test should therefore fail (as it did). So the fix here was indeed necessary and not incorrect. :)